app/(candidate)/dashboard/tasks/components/TaskItem.js

```diff
@@ -39,42 +25,22 @@ export default function TaskItem({
   } = task
 
   const handleAction = () => {
-    if (proRequired && !isPro) {
-      // TODO: direct to pro upgrade
-      return
-    }
+    onAction(task)
+  }
 
-    switch (flowType) {
-      case TASK_TYPES.texting:
-        // TODO: implement texting flow
-        console.log('texting flow')
-        break
-      case TASK_TYPES.robocall:
-        // TODO: implement robocall flow
-        console.log('robocall flow')
-        break
-      case TASK_TYPES.doorKnocking:
-        // TODO: implement door knocking flow
-        console.log('door knocking flow')
-        break
-      case TASK_TYPES.phoneBanking:
-        // TODO: implement phone banking flow
-        console.log('phone banking flow')
-        break
-      default:
-        console.warn('Unknown task type:', flowType)
-    }
+  const handleCheck = () => {
+    onCheck(task)
   }
 
-  const isExternalLink = flowType === TASK_TYPES.link
+  const isExternalLink = link !== undefined
```

Comment:
```suggestion
  const isExternalLink = Boolean(link)
```

Instead of using explicit `undefined`?